### PR TITLE
Migrate publication of docs and webstart files to nexus server

### PR DESCRIPTION
**Pull request by lesserwhirls**

### CODE REVIEW

File: docs/internal/release.md [hunk 1]
```diff
@@ -101,119 +100,86 @@
...
+      * `./gradlew clean :ui:publishWebstart`
+
+17. Release documentation and javadocs to nexus
+    - remove old 4.6 docs
+      * `./gradlew -PdryRun=false :docs:website:deleteAllDocs`
+    - upload new 4.6 docs
+      * `./gradlew :docs:website:publishAllDocs`
+
+18. Update Unidata download page(s)
     - check https://www.unidata.ucar.edu/downloads/tds/
```
Comment (oxelson) [inline docs/internal/release.md:118]: Let's remove the reference to toc.xml as the downloads isn't used anymore for the THREDDS project.

File: docs/website/netcdf-java/index.html [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  <li><a href="reference/Cookbook.html">Cookbook</a></li>
+  <li><a href="/software/netcdf/examples/programs/">Example Basic Programs in C/C++/Fortran/Java</a></li>
+  <li><a href="reference/index.html">Reference Documentation </a></li>
+  <li><a href="CDM/index.html">Common Data Model</a></li>
+  <li><a href="ncml/index.htm">NetCDF Markup Language</a>: <a
+      href="https://www.unidata.ucar.edu/software/netcdf/ncml/AnnotatedSchema4.html">annotated schema</a></li>
+</ul>
+<h3>CF Conventions</h3>
+<ul>
+  <li><a href="http://thredds-test.unidata.ucar.edu/thredds/cf_examples.html">Sample CF DSG Files</a></li>
```
Comment (oxelson) [inline docs/website/netcdf-java/index.html:112]: http:// --> https://

File: docs/website/netcdf-java/ncml/Tutorial.html [hunk 3]
```diff
@@ -30,7 +30,7 @@ <h2>Introduction</h2>
   of the functionality in the CDM library. You may also use a specialized XML editor like Oxygen or XML Spy, or an IDE
   like IntelliJ, Eclipse, or NetBeans that have context-sensitive XML editing. </p>
 
-<p>For these exercises, you should <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">run ToolsUI from
+<p>For these exercises, you should <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">run ToolsUI from
```
Comment (oxelson) [inline docs/website/netcdf-java/ncml/Tutorial.html:33]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

Comment (oxelson) [inline docs/website/netcdf-java/ncml/Tutorial.html:33]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/netcdf-java/ncml/index.htm [hunk 4]
```diff
@@ -38,7 +38,7 @@ <h3>NcML as output </h3>
 <p>The simplest use of NcML is to describe the metadata and structural content of a netCDF file, like the output of <strong>ncdump -h</strong>. There are
   several ways to generate NcML ouput from an existing netCDF file:</p>
 <ol>
-  <li>In the <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a> application, use the NcML tab to open a
+  <li>In the <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a> application, use the NcML tab to open a
```
Comment (oxelson) [inline docs/website/netcdf-java/ncml/index.htm:41]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/netcdf-java/ncml/index.htm [hunk 5]
```diff
@@ -53,7 +53,7 @@ <h3>Using NcML to create a NetCDF-3 file </h3>
 <ol>
   <li>Using a text or XML editor, create the NcML file with an &quot;xml&quot; or &quot;ncml&quot; file extension. Open the NcML file with <strong>NetcdfDataset.open()</strong>,
     then call <strong>ucar.nc2.FileWriter.writeFile()</strong></li>
-  <li>In the <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a> application, open an existing NcML file in
+  <li>In the <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a> application, open an existing NcML file in
```
Comment (oxelson) [inline docs/website/netcdf-java/ncml/index.htm:56]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/netcdf-java/ncml/index.htm [hunk 6]
```diff
@@ -71,7 +71,7 @@ <h3>Using NcML to modify<a name="modify"></a> an existing CDM file </h3>
         write the new file, eg:
         <pre><em><strong></strong>java -Xmx1g -classpath netcdfAll-4.3.jar <strong>ucar.nc2.dataset.NetcdfDataset -in myFile.ncml -out myFile.nc</strong></em></pre>
       </li>
-      <li> (or) In the <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a> application, open the NcML file in
+      <li> (or) In the <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a> application, open the NcML file in
```
Comment (oxelson) [inline docs/website/netcdf-java/ncml/index.htm:74]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/netcdf-java/reference/FeatureDatasets/CFpointImplement.html [hunk 7]
```diff
@@ -131,7 +131,7 @@ <h3>Step 4: Test a sample file</h3>
     <p>Write a test file with some sample data in it.</p>
   </li>
   <li>
-    <p>Test in ToolsUI 4.3 (<a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">webstart link</a>)</p>
+    <p>Test in ToolsUI 4.3 (<a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">webstart link</a>)</p>
```
Comment (oxelson) [inline docs/website/netcdf-java/reference/FeatureDatasets/CFpointImplement.html:134]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/netcdf-java/tutorial/NetcdfFile.html [hunk 8]
```diff
@@ -43,7 +43,7 @@ <h2 >Opening a <a>NetcdfFile</a>
 <h3>Using ToolsUI to browse the metadata of a dataset </h3>
 
 <p>The NetCDF Tools User Interface (aka ToolsUI) is a program for browsing and debugging NetCDF files. You can start it from the command line, or
-  <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">launch it from webstart</a>. </p>
+  <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">launch it from webstart</a>. </p>
```
Comment (oxelson) [inline docs/website/netcdf-java/tutorial/NetcdfFile.html:46]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/reference/NetcdfSubsetServiceConfigure.html [hunk 9]
```diff
@@ -48,14 +48,14 @@ <h2><a name="restrictions">Restrictions on what files can be served</a></h2>
...
-    following <a href="/software/netcdf-java/v4.6/reference/FeatureDatasets/Overview.html">Feature Types</a>: GRID, POINT,
+    following <a href="/netcdf-java/4.6/userguide/reference/FeatureDatasets/Overview.html">Feature Types</a>: GRID, POINT,
     STATION.
 </p>
 
 <h3>Verifying that a dataset is gridded</h3>
 
 <p>Open your file in the
-    <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">NetCDF ToolsUI
+    <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">NetCDF ToolsUI
```
Comment (oxelson) [inline docs/website/tds/reference/NetcdfSubsetServiceConfigure.html:58]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/reference/WCS.html [hunk 10]
```diff
@@ -23,7 +23,7 @@ <h2>Which files can be served through the WCS server? </h2>
   <li>Data files must contain gridded data.</li>
   <li>The
     NetCDF-Java Common Data Model must be able to identify the coordinates
-    system used. Check this by opening in the Grid Panel of the <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a>
+    system used. Check this by opening in the Grid Panel of the <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a>
```
Comment (oxelson) [inline docs/website/tds/reference/WCS.html:26]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/reference/NetcdfSubsetServiceConfigure.html (hunk 9 above)
Comment (oxelson) [inline docs/website/tds/reference/NetcdfSubsetServiceConfigure.html:58]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/reference/WMS.html [hunk 11]
```diff
@@ -25,7 +25,7 @@ <h2><a name="Which_files_can_be_served_through_the"></a>Which files can be serve
   <li>Data files must contain gridded data.</li>
   <li>The NetCDF-Java Common Data Model must be able to identify the coordinates
     system used. Check this by opening in the Grid Panel of the
-    <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a>
+    <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a>
```
Comment (oxelson) [inline docs/website/tds/reference/WMS.html:28]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/tutorial/CatalogPrimer.html [hunk 12]
```diff
@@ -436,7 +436,7 @@ <h2><a name="Using_ToolsUI">Using ToolsUI to view catalogs</a></h2>
 
     <p>The NetCDF Tools User Interface (aka ToolsUI) can read and display THREDDS catalogs. You can start it from the
         command line, or
-        <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">launch it
+        <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">launch it
```
Comment (oxelson) [inline docs/website/tds/tutorial/CatalogPrimer.html:439]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/tutorial/ConfigCatalogs.html [hunk 13]
```diff
@@ -141,12 +141,12 @@ <h4>Data Requirement for Each Service</h4>
 
   <p>You can check that a data file is recognized as "gridded" with netCDF-Java ToolsUI (available for download from the
     <a href="https://www.unidata.ucar.edu/software/netcdf-java/">netCDF-Java home page</a> or you can use
-    <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">webstart</a>).</p>
+    <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">webstart</a>).</p>
```
Comment (oxelson) [inline docs/website/tds/tutorial/ConfigCatalogs.html:144]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/tutorial/ConfigCatalogs.html [hunk 14]
```diff
@@ -141,12 +141,12 @@ <h4>Data Requirement for Each Service</h4>
...
 
   <div id="expanded" class="exercise">
     <h3>Exercise: Check that the NAM Dataset is Gridded Data</h3>
     <ol>
-      <li>Open the netCDF-Java ToolsUI application [<a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">webstart</a>].
+      <li>Open the netCDF-Java ToolsUI application [<a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">webstart</a>].
```
Comment (oxelson) [inline docs/website/tds/tutorial/ConfigCatalogs.html:149]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/tutorial/NcML.htm [hunk 15]
```diff
@@ -335,10 +335,10 @@ <h2><a name="debug">Debugging NcML</a></h2>
         <li>Remove the <code>recheckEvery</code> attribute if present on the <code>&lt;scan&gt;</code> element.</li>
         <li>Make sure that the <code>&lt;scan&gt;</code> location is available on the machine you are running ToolsUI</li>
       </ol>
-      Now start up <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/webstart/netCDFtools.jnlp">ToolsUI</a>, and in the viewer tab, navigate to
+      Now start up <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/webstart/netCDFtools.jnlp">ToolsUI</a>, and in the viewer tab, navigate to
```
Comment (oxelson) [inline docs/website/tds/tutorial/NcML.htm:338]: Let's just reference the URL on the docs server directly and not rely on the redirect we will put into place.

File: docs/website/tds/tutorial/Subset.html [hunk 16]
```diff
@@ -253,8 +253,8 @@ <h2><a name="CDMRemote">CDMRemote</a></h2>
 
 <p>
 <li><strong>Reference</strong>:
-<a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/reference/stream/CdmRemote.html target="_blank">
-https://www.unidata.ucar.edu/software/netcdf-java/v4.6/reference/stream/CdmRemote.html</a>
+<a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/reference/stream/CdmRemote.html target="_blank">
```
Comment (oxelson) [inline docs/website/tds/tutorial/Subset.html:256]: Spoke with Doug about this.  Let's use the convention of https://www.unidata.ucar.edu/**software**/<package_name>/*

File: docs/website/tds/tutorial/NcML.htm [hunk 17]
```diff
@@ -335,10 +335,10 @@ <h2><a name="debug">Debugging NcML</a></h2>
...
       <code>test.ncml</code> and try to open it.
     </li>
-    <li>If the dataset is <a href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/ncml/Aggregation.html#dynamic">dynamic</a> (files can be added or deleted),
+    <li>If the dataset is <a href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/ncml/Aggregation.html#dynamic">dynamic</a> (files can be added or deleted),
```
Comment (oxelson) [inline docs/website/tds/tutorial/NcML.htm:341]: Convert to relative links?

File: docs/website/netcdf-java/reference/netcdf4Clibrary.html [hunk 18]
```diff
@@ -153,7 +153,7 @@ <h3><a name="loading_alternate" id="loading_alternate">Alternate methods</a></h3
             <li>set the library path and name in the <a href="RuntimeLoading.html#XML">runtime configuration file</a>
             </li>
             <li>directly call <a
-                    href="https://www.unidata.ucar.edu/software/netcdf-java/v4.6/javadocAll/ucar/nc2/jni/netcdf/Nc4Iosp.html#setLibraryAndPath(java.lang.String,%20java.lang.String)">Nc4Iosp.setLibraryAndPath()</a>
+                    href="https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/javadocAll/ucar/nc2/jni/netcdf/Nc4Iosp.html#setLibraryAndPath(java.lang.String,%20java.lang.String)">Nc4Iosp.setLibraryAndPath()</a>
```
Comment (oxelson) [inline docs/website/netcdf-java/reference/netcdf4Clibrary.html:156]: Reference javadocs on docs server?

Comment (lesserwhirls) [reply docs/website/netcdf-java/reference/netcdf4Clibrary.html:156]: Yep - missed grabbing those.

File: docs/website/tds/tutorial/NcML.htm (hunk 17 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/NcML.htm:341]: Totally.

File: docs/website/tds/tutorial/Subset.html (hunk 16 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/Subset.html:256]: > https://www.unidata.ucar.edu/netcdf-java/4.6/userguide/reference/stream/CdmRemote.html

Oops, this should actually be https://docs.unidata.ucar.edu/netcdf-java/4.6/userguide/reference/stream/CdmRemote.html, or better yet `/../../reference/stream/CdmRemote.html`

**Review by oxelson — CHANGES_REQUESTED** [review]
Looks good @lesserwhirls!  Let's chat about the suggestions this afternoon.

File: docs/internal/release.md (hunk 1 above)
Comment (lesserwhirls) [reply docs/internal/release.md:118]: 👍

File: docs/website/netcdf-java/index.html (hunk 2 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/index.html:112]: 👍

File: docs/website/netcdf-java/ncml/Tutorial.html (hunk 3 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/ncml/Tutorial.html:33]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/netcdf-java/ncml/index.htm (hunk 4 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/ncml/index.htm:41]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/netcdf-java/ncml/index.htm (hunk 5 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/ncml/index.htm:56]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/netcdf-java/reference/FeatureDatasets/CFpointImplement.html (hunk 7 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/reference/FeatureDatasets/CFpointImplement.html:134]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/reference/WCS.html (hunk 10 above)
Comment (lesserwhirls) [reply docs/website/tds/reference/WCS.html:26]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/reference/NetcdfSubsetServiceConfigure.html (hunk 9 above)
Comment (lesserwhirls) [reply docs/website/tds/reference/NetcdfSubsetServiceConfigure.html:58]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/tutorial/CatalogPrimer.html (hunk 12 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/CatalogPrimer.html:439]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/tutorial/ConfigCatalogs.html (hunk 14 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/ConfigCatalogs.html:149]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/tutorial/NcML.htm (hunk 17 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/NcML.htm:341]: Moved to relative (or absolute path for cross doc) links

File: docs/website/netcdf-java/reference/netcdf4Clibrary.html (hunk 18 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/reference/netcdf4Clibrary.html:156]: Fixed

File: docs/website/tds/tutorial/ConfigCatalogs.html (hunk 13 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/ConfigCatalogs.html:144]: Fixed

File: docs/website/tds/reference/WMS.html (hunk 11 above)
Comment (lesserwhirls) [reply docs/website/tds/reference/WMS.html:28]: Fixed

File: docs/website/netcdf-java/ncml/index.htm (hunk 6 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/ncml/index.htm:74]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/netcdf-java/tutorial/NetcdfFile.html (hunk 8 above)
Comment (lesserwhirls) [reply docs/website/netcdf-java/tutorial/NetcdfFile.html:46]: webstart links updated to point to downloads.unidata.ucar.edu.

File: docs/website/tds/tutorial/NcML.htm (hunk 15 above)
Comment (lesserwhirls) [reply docs/website/tds/tutorial/NcML.htm:338]: webstart links updated to point to downloads.unidata.ucar.edu.

**Comment by lesserwhirls** [comment]
Ok, should be updated now to reflect changes related to the comments - great catches, btw! So many crufty URLs, and one or more find/replace that didn't work like I thought :-/

**Review by oxelson — APPROVED** [review]

**Comment by oxelson** [comment]
> Ok, should be updated now to reflect changes related to the comments - great catches, btw! So many crufty URLs, and one or more find/replace that didn't work like I thought :-/

We're bound to miss something with all of the changes made.  I'm sure there a few we didn't catch.  But, we'll fix them as we come across them.  Nice work @lesserwhirls !

**Comment by lesserwhirls** [comment]
Thank you @oxelson!

**Pull request merged**